File: src/net/fodt/Spec/Text/List.cs
// FormatFactory.Fodt — Spec.Text.List — Canonical spec-shaped model class
// spec_qname: text:list
// spec_fact_ref: FACT-FODT-005
// TC-QHARD-051: converted from architecture_only stub to real model class
namespace FormatFactory.Fodt.Spec.Text;

/// <summary>
/// Spec-shaped model class for the ODF text:list element.
///
/// ODF 1.3 §5.3.1 — text:list is the container element for list items in an
/// ODF text document. It contains text:list-item children.
/// spec_qname: text:list
/// spec_fact_ref: FACT-FODT-005
/// </summary>
public sealed class List
{
    /// <summary>The ODF QName for this element. Grounded in ODF 1.3 §5.3.1.</summary>
    public const string SpecQName = "text:list";

    /// <summary>The SAL fact reference for this element (sal-facts-latest.json).</summary>
    public const string SpecFactRef = "FACT-FODT-005";

    /// <summary>Number of text:list-item children.</summary>
    public int ItemCount { get; init; }

    /// <summary>The text:style-name attribute — list style applied. May be null.</summary>
    public string? StyleName { get; init; }
}

File: src/python/fodt/spec/text/list_.py
from __future__ import annotations
from typing import Any, ClassVar


class List:
    """Canonical spec-shaped class for text:list in FODT context (ODF §5.3.1)."""

    spec_qname: ClassVar[str] = "text:list"
    spec_fact_ref: ClassVar[str] = "FACT-FODT-005"

    def __init__(self, data: dict[str, Any]):
        self._data = data

    @property
    def items(self) -> list:
        return self._data.get("items", [])

    @property
    def style_name(self) -> str:
        return self._data.get("style_name", "")

    def to_dict(self) -> dict[str, Any]:
        return dict(self._data)
